tests/testthat/test-growth_rate.R

```diff
@@ -0,0 +1,14 @@
+library(dplyr)
+
+x <- jhu_csse_daily_subset %>%
+  select(-case_rate_7d_av,-death_rate_7d_av, -cases_7d_av) %>%
+  group_by(geo_value) %>% 
+  mutate(cases_gr = growth_rate(x = time_value,  y = cases))
+
+test_that("Test error throwing",{
+  expect_error(growth_rate(x=1:3,y=1:4),
+               "`x` and `y` must have the same length.")
+  expect_error(growth_rate(x=1:20,y=1:20,x0=21),
+               "`x0` must be a subset of `x`.")
+  expect_error(growth_rate(x=1:20,y=1:20,x0=c(1,3)),NA)
+})
```

Comment:
Could you please add another test for the actual values coming out?  E.g., construct a simple example where the growth rate should be obvious, and test that we calculate that obvious value.  At least for some of the `method`s.

Also, please test that each of the `method`s, `log_scale` settings, and `na_rm` settings at least run successfully.  If you can't construct an example where you think you know what the output should be, maybe throw in some `testthat::expect_snapshot`s?

Reply:
(I assume we'll need to take the `expect_snapshot` approach for some/most/all of the `method`s; I'm not sure, e.g., that there's a simple example where we know the trendfilter output.)

Reply:
`expect_snapshot` creates a new file for snapshots and does something that doesn't seem inherently helpful. All it does it checks for errors from my experience; however, that doesn't seem to be any better than doing `expect_error`.

Reply:
@kenmawer I think the point here is to test that the correct calculations are occurring when arguments are correctly specified, not just that errors happen when the inputs are wrong.

Reply:
I already added a few tests for calculations. I would suspect that for some of the other methods, it would be too complicated to be able to verify the outputted numbers. It also depends on what you mean by "correct calculations," whether it's merely getting the right number of cells or getting those exact numbers.

Reply:
Since it's likely too complicated to verify the output numbers, that's why I suggested `expect_snapshot`.  When you first run it, it saves the current output to a _snaps directory, which you would then commit (make sure that the test is constructed so that this file is very small; we don't want to bloat the repo), and then later, when we run the snapshot test again, it will detect the file in the _snaps directory and compare against it.  If we inadvertently change the output behavior, this test will catch it.

Reply:
Looks like of the requested

> test that each of the methods, log_scale settings, and na_rm settings at least run successfully

only the method outputs are left -- is that correct?